core/trino-main/src/main/resources/webapp/src/components/WorkerList.jsx

```diff
@@ -0,0 +1,140 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from "react";
+
+import {
+    addToHistory,
+} from "../utils";
+
+const SMALL_SPARKLINE_PROPERTIES = {
+    width: '100%',
+    height: '57px',
+    fillColor: '#3F4552',
+    lineColor: '#747F96',
+    spotColor: '#1EDCFF',
+    tooltipClassname: 'sparkline-tooltip',
+    disableHiddenCheck: true,
+};
+
+export class WorkerList extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            workerInfo: null,
+            initialized: false,
+            ended: false,
+
+            workerId: [],
+            workerIp: [],
+            workerVersion: [],
+            coordinator: [],
+            state: [],
+        };
+        this.refreshLoop = this.refreshLoop.bind(this);
+    }
+
+    refreshLoop() {
+        clearTimeout(this.timeoutId); // to stop multiple series of refreshLoop from going on simultaneously
+        // const nodeId = getFirstParameter(window.location.search);
+        $.get('/ui/api/worker', function (workerInfo) {
+            this.setState({
+                initialized: true,
+                workerInfo: workerInfo
+            })
+            for (var index in workerInfo) {
+                this.setState({
```

Comment:
I think that you didn't design the state well. Instead of
```
{
            workerId: string[],
            workerIp: string[],
            workerVersion: string[],
            ...
}
```
I would do (typescript notation):
```
{
   workers: [
            workerId: string,
            workerIp: string,
            workerVersion: string,
            ...
   ]
}
```

This way, the model is more concise and easier to work on

Reply:
done.